supernode/daemon/mgr/dfgettask/manager_test.go

```diff
@@ -189,6 +190,204 @@ func (s *DfgetTaskMgrTestSuite) TestDfgetTaskUpdate(c *check.C) {
+		c.Check(err != nil, check.Equals, tc.expectedError)
+		c.Check(got, check.Equals, tc.expected)
+	}
+}
+
+func (s *DfgetTaskMgrTestSuite) TestDfgetTaskGetCIDsByTaskID(c *check.C) {
+	manager, _ := NewManager(s.cfg, prometheus.NewRegistry())
+	// peer1
+	manager.Add(context.Background(), &types.DfGetTask{
+		CID:        "foo",
+		CallSystem: "foo",
+		Dfdaemon:   false,
+		Path:       "/peer/file/taskFileName",
+		PieceSize:  4 * 1024 * 1024,
+		TaskID:     "test1",
+		PeerID:     "peer1",
+	})
+	// peer2
+	manager.Add(context.Background(), &types.DfGetTask{
+		CID:        "foo",
+		CallSystem: "foo",
+		Dfdaemon:   true,
+		Path:       "/peer/file/taskFileName",
+		PieceSize:  4 * 1024 * 1024,
+		TaskID:     "test1",
+		PeerID:     "peer2",
+	})
+	manager.Add(context.Background(), &types.DfGetTask{
+		CID:        "bar",
+		CallSystem: "bar",
+		Dfdaemon:   true,
+		Path:       "/peer/file/taskFileName",
+		PieceSize:  4 * 1024 * 1024,
+		TaskID:     "test2",
+		PeerID:     "peer2",
+	})
+
+	cases := []struct {
+		taskID        string
+		expected      []string
+		expectedError bool
+	}{
+		{
+			taskID:   "",
+			expected: nil,
+		},
+		{
+			taskID:   "test1",
+			expected: []string{"foo", "foo"},
+		},
+		{
+			taskID:   "test2",
+			expected: []string{"bar"},
+		},
+	}
+
+	for _, tc := range cases {
+		got, err := manager.GetCIDsByTaskID(context.Background(), tc.taskID)
+		c.Check(err != nil, check.Equals, tc.expectedError)
+		c.Check(reflect.DeepEqual(got, tc.expected), check.Equals, true)
+	}
+}
+
+func (s *DfgetTaskMgrTestSuite) TestDfgetTaskGetCIDAndTaskIDsByPeerID(c *check.C) {
+	manager, _ := NewManager(s.cfg, prometheus.NewRegistry())
+	// peer1
+	manager.Add(context.Background(), &types.DfGetTask{
+		CID:        "foo",
+		CallSystem: "foo",
+		Dfdaemon:   false,
+		Path:       "/peer/file/taskFileName",
+		PieceSize:  4 * 1024 * 1024,
+		TaskID:     "test1",
+		PeerID:     "peer1",
+	})
+	manager.Add(context.Background(), &types.DfGetTask{
+		CID:        "foo",
+		CallSystem: "foo",
+		Dfdaemon:   false,
+		Path:       "/peer/file/taskFileName",
+		PieceSize:  4 * 1024 * 1024,
+		TaskID:     "test2",
+		PeerID:     "peer1",
+	})
+	// peer2
+	manager.Add(context.Background(), &types.DfGetTask{
+		CID:        "foo",
+		CallSystem: "foo",
+		Dfdaemon:   true,
+		Path:       "/peer/file/taskFileName",
+		PieceSize:  4 * 1024 * 1024,
+		TaskID:     "test1",
+		PeerID:     "peer2",
+	})
+	manager.Add(context.Background(), &types.DfGetTask{
+		CID:        "bar",
+		CallSystem: "bar",
+		Dfdaemon:   true,
+		Path:       "/peer/file/taskFileName",
+		PieceSize:  4 * 1024 * 1024,
+		TaskID:     "test2",
+		PeerID:     "peer2",
+	})
+
+	cases := []struct {
+		peerID        string
+		expected      map[string]string
+		expectedError bool
+	}{
+		{
+			peerID:   "",
+			expected: map[string]string{},
+		},
+		{
+			peerID: "peer1",
+			expected: map[string]string{
+				"foo": "test2",
+			},
```

Comment:
Is the `cID` of the `task` on the same `peer` unique?

Reply:
Yes, you can find the definition of CID at `apis/types/df_get_task.go#DfGetTask`. 

```
// CID means the client ID. It maps to the specific dfget process.
// When user wishes to download an image/file, user would start a dfget process to do this.
// This dfget is treated a client and carries a client ID.
// Thus, multiple dfget processes on the same peer have different CIDs.
```